libtrip-notification/src/test/java/com/mapbox/navigation/trip/notification/MapboxTripNotificationTest.kt

```diff
@@ -124,4 +136,145 @@ class MapboxTripNotificationTest {
 
         verify(exactly = 1) { NavigationNotificationProvider.buildNotification(any()) }
     }
+
+    @Test
+    fun whenUpdateNotificationCalledThenBannerInstructionsPrimaryTextInteracted() {
+        val routeProgress = mockk<RouteProgress>(relaxed = true)
+        val primaryText = { "Primary Text" }
+        val bannerText = mockBannerText(routeProgress, primaryText)
+        mockUpdateNotificationAndroidInteractions()
+
+        notification.updateNotification(routeProgress)
+
+        verify(exactly = 1) { bannerText.text() }
+        verify(exactly = 2) { anyConstructed<RemoteViews>().setTextViewText(any(), primaryText()) }
```

Comment:
Instead of matching the whole class, you can leverage something like `NavigationNotificationProvider` that will provide the `RemoteView` for you. You can mock it as you do in other tests, which will give you a chance to verify if we're setting the correct text on the desired view rather than if we are setting it at all anywhere.

Reply:
Yeah,  firstly I didn't want to touch an initial implementation at all.
Now it's already touched, so I'll make some improvements

Reply:
Ah okay, got it.